akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -27,6 +27,14 @@ private class BackoffOnRestartSupervisor(
 
   import context._
   import BackoffSupervisor._
+
+  if (strategy.withinTimeRange.isFinite()) {
+    // If the user has defined a time range for the maxNrOfRetries, we'll queue up a message
+    // to ourselves every time that range elapses, to reset the restart counter.
+    val finiteWithinTimeRange = strategy.withinTimeRange.asInstanceOf[FiniteDuration]
+    system.scheduler.schedule(finiteWithinTimeRange, finiteWithinTimeRange, self, ResetRestartCount(restartCount))
+  }
```

Comment:
Hmm. So this will actually keep sending messages even though we are not at all restarting, wouldn't a solution that only does processing when there was a restart be better?